### PR TITLE
Keras Tensor NP Functions

### PR DESCRIPTION
**Pull request by kamathis4**
Implemented functions as per TODO in Keras tensor
[Issue](https://github.com/keras-team/keras-core/issues/518)

### CODE REVIEW

File: keras_core/ops/numpy.py [hunk 1]
```diff
@@ -3457,3 +3457,21 @@ def compute_output_spec(self, N, M=None, k=0, dtype="float32"):
...
+
+    def compute_output_spec(self, x1, x2):
+        x1_shape = getattr(x1, "shape", [])
+        x2_shape = getattr(x2, "shape", [])
+        output_shape = broadcast_shapes(x1_shape, x2_shape)
+        return KerasTensor(output_shape, dtype=x1.dtype)
+
+
+@keras_core_export(["keras_core.ops.floor", "keras_core.ops.numpy.floor"])
+def floordiv(x1, x2):
```
Comment (fchollet) [inline keras_core/ops/numpy.py:3474]: If it's exported as `floor` then it should be called `floor` and the op class should be `Floor`.

Comment (kamathis4) [reply keras_core/ops/numpy.py:3474]: Changed export to floordiv

**Comment by kamathis4** [comment]
Do you want exactly the __eq__ operator to work? As in two tensors are equal or can I work on the equality of values as specified in the **keras_core.ops.equal**

**Comment by kamathis4** [comment]
So to reiterate, I shouldn't implement the __eq__, instead raise an exception?

**Comment by fchollet** [comment]
Correct!

**Comment by kamathis4** [comment]
I think in multiple tests we are asserting equality between tests and thus the raise error is failing.

File: keras_core/backend/jax/numpy.py [hunk 2]
```diff
@@ -573,3 +573,11 @@ def sum(x, axis=None, keepdims=False):
 
 def eye(N, M=None, k=0, dtype="float32"):
     return jnp.eye(N, M=M, k=k, dtype=dtype)
+
+
+def floordiv(x1, x2):
```
Comment (fchollet) [inline keras_core/backend/jax/numpy.py:578]: We follow the NumPy API, so this should be called `floor_divide`.

File: keras_core/backend/common/keras_tensor.py [hunk 3]
```diff
@@ -201,32 +201,99 @@ def __rpow__(self, other):
...
+        from keras_core import ops
+
+        return ops.GreaterEqual().symbolic_call(self, other)
+
+    def __ne__(self, other):
+        from keras_core import ops
+
+        return ops.NotEqual().symbolic_call(self, other)
+
+    def __eq__(self, other):
```
Comment (fchollet) [inline keras_core/backend/common/keras_tensor.py:249]: Just remove this method altogether I guess.

**Review by fchollet — COMMENTED** [review]
Thanks for the updates!

File: keras_core/backend/common/keras_tensor.py [hunk 4]
```diff
@@ -201,32 +201,99 @@ def __rpow__(self, other):
...
-    #   "__eq__",
-    #   "__and__",
-    #   "__rand__",
-    #   "__or__",
-    #   "__ror__",
-    #   "__xor__",
-    #   "__rxor__",
-    #   "__invert__",
     #   "broadcast_to"
     #   "astype"
```
Comment (fchollet) [inline keras_core/backend/common/keras_tensor.py:296]: We likely don't need the last 2.

**Comment by kamathis4** [comment]
Any suggestions to fix the import part @fchollet ?

File: keras_core/backend/jax/numpy.py [hunk 5]
```diff
@@ -573,3 +573,11 @@ def sum(x, axis=None, keepdims=False):
 
 def eye(N, M=None, k=0, dtype="float32"):
     return jnp.eye(N, M=M, k=k, dtype=dtype)
+
+
+def floor_divide(x1, x2):
```
Comment (fchollet) [inline keras_core/backend/jax/numpy.py:578]: Please add these 2 functions to `backend/numpy/numpy.py` as well.

**Pull request closed without merging**